src/greynirseq/nicenlp/tasks/document_translation_from_pretrained_bart.py

```diff
@@ -0,0 +1,385 @@
+
+        bpe = SentencepieceBPE(SentencepieceConfig(sentencepiece_model=self.cfg.spm_model))
+        noisy_bpe = SentencepieceBPE(
+            SentencepieceConfig(sentencepiece_model=self.cfg.spm_model, sentencepiece_enable_sampling=True)
+        )
+        from greynirseq.nicenlp.data.encoders import Encoder
+
+        my_enc = Encoder(
+            dictionary=self.the_dict,
+            bpe=bpe,
+            noisy_bpe=noisy_bpe,
+            allowed_dictionary_min=self.the_dict.nspecial,
+            allowed_dictionary_max=len(self.the_dict) - 1 - len(self.langs),  # type: ignore
+            fragment_noise_prob=self.cfg.fragment_noise_prob,
+            global_skip_noise_prob=self.cfg.global_skip_noise_prob,
+            word_noise_config=self.cfg.word_noise_config,
+            char_noise_config=self.cfg.char_noise_config,
+            max_sequence_length=self.cfg.max_sequence_length,
+        )
+
+        def decode(example):
+            src_string = self.the_dict.string(example["source"])
+            tgt_string = self.the_dict.string(example["target"])
+            # decoded = bpe.decode(spm_string)
+            print()
+            print(bpe.decode(src_string))
+            print()
+            print(bpe.decode(tgt_string))
+            print()
+
+        datasets = []
+        for _, dataset_values in datasets_for_loading.items():
+            dataset = IndexedParallelDocumentsDataset.from_parallel_jsonl(
+                name=dataset_values["name"],
+                is_bt=dataset_values["is_bt"],
+                src_path=dataset_values["src_path"],
+                tgt_path=dataset_values["tgt_path"],
+                bpe_encoder=bpe,
+                dictionary=self.the_dict,
+                encoder=my_enc,
+                data_language_mapper=self.language_mappings,
+                max_seq_len=self.cfg.max_sequence_length,
+                max_merges=self.cfg.max_merges,
+                align_path=dataset_values["align_path"],
+                num_proc=self.cfg.num_preprocess_workers,
+                seed=self.cfg.seed,
+            )
+            datasets.append(dataset)
+
+        if len(datasets) != 1:
+            parallel_datasets = [dataset for dataset in datasets if not dataset.is_bt]
+            bt_datasets = [dataset for dataset in datasets if dataset.is_bt]
+
+            dataset = IndexedParallelBTDocumentsDataset(
+                parallel_datasets,
+                bt_datasets,
+                self.the_dict,
+                encoder=my_enc,
+                parallel_prob=self.cfg.parallel_prob,
+                seed=self.cfg.seed,
+                max_seq_len=self.cfg.max_sequence_length,
+                max_merges=self.cfg.max_merges,
+                num_proc=self.cfg.num_preprocess_workers,
+                no_merge_prob=self.cfg.no_merge_prob,
+            )
+        else:
+            dataset = datasets[0]
+
+        dataset.set_epoch(1)
```

Comment:
Add comment that this can take a while since it causes an interleave datasets call.